src/Bindings/RabbitMQAsyncCollector.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+
+namespace Microsoft.Azure.WebJobs.Extensions.RabbitMQ
+{
+    public class RabbitMQAsyncCollector : IAsyncCollector<string>
+    {
+        private readonly RabbitMQContext context;
+        private readonly ConcurrentQueue<string> messages = new ConcurrentQueue<string>();
+        private readonly ConnectionFactory factory;
+        private readonly IConnection connection;
+        private readonly IModel channel;
+        private readonly QueueDeclareOk queue;
+
+        public RabbitMQAsyncCollector(RabbitMQContext context)
+        {
+            this.context = context;
+            this.factory = new ConnectionFactory() { HostName = context.Hostname };
+            this.connection = this.factory.CreateConnection();
+            this.channel = this.connection.CreateModel();
+            this.queue = CreateQueue(this.channel, this.context);
+        }
+
+        public Task AddAsync(string message, CancellationToken cancellationToken = default)
+        {
+            message = this.context.Message;
+            this.messages.Enqueue(message);
+
+            return Task.CompletedTask;
+        }
+
+        public async Task FlushAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            while (this.messages.TryDequeue(out string message))
+            {
+                await this.PublishAsync(message);
+            }
+        }

```

Comment:
Instead of adding messages to our own queue, we should try using `BasicPublishBatch` -- I bet there's a lot of optimizations they do that would make this faster: http://rabbitmq.github.io/rabbitmq-dotnet-client/api/RabbitMQ.Client.Impl.BasicPublishBatch.html. Looks like you can call `model.CreateBasicPublishBatch()`?

Reply:
I feel like this functionality might be deprecated? I could be wrong but I can't find any other documentation and the Github links are broken